tests/test_page.py

```diff
@@ -248,6 +248,40 @@ def test_latest_version_does_not_add_noindex_for_robots(
     assert '<meta name="robots" content="noindex">' not in str(resp.data)
 
 
+def test_latest_version_does_not_add_noindex_for_robots_when_newer_draft_exists(
+    app,
+    db,
+    db_session,
+    test_app_client,
+    mock_admin_user,
+    stub_topic_page,
+    stub_subtopic_page,
+    stub_measure_page_one_of_three,
+    stub_measure_page_two_of_three,
+    stub_measure_page_three_of_three,
+):
+    # GIVEN the latest version of a page and a newer draft (ie the stub_measure_page_three_of_three fixture)
+    latest_published_version_of_page = stub_measure_page_two_of_three
+
+    # WHEN we get the rendered template
+    with test_app_client.session_transaction() as session:
+        session["user_id"] = mock_admin_user.id
+    from flask import url_for
+
+    resp = test_app_client.get(
+        url_for(
+            "static_site.measure_page",
+            topic=stub_topic_page.uri,
+            subtopic=stub_subtopic_page.uri,
+            measure=latest_published_version_of_page.uri,
+            version=latest_published_version_of_page.version,
+        )
+    )
+    # THEN it should not contain a noindex tag
+    assert resp.status_code == 200
+    assert '<meta name="robots" content="noindex">' not in str(resp.data)
```

Comment:
I might be tempted to use beautifulsoup to parse the html here, and assert that there's no `meta[name=robots]` element containing `noindex` in the `content` attribute, so that it doesn't false positive if other keywords get added, eg `<meta name="robots" content="noindex,noarchive">`?

Reply:
Done.